src/parser/xasl_generation.c

```diff
@@ -7545,7 +7545,7 @@ pt_to_regu_variable (PARSER_CONTEXT * parser, PT_NODE * node, UNBOX unbox)
       CAST_POINTER_TO_NODE (node);
 
       if (node != NULL && node->type_enum == PT_TYPE_LOGICAL
-	  && (node->node_type == PT_EXPR || node->node_type == PT_VALUE))
+	  && (node->node_type == PT_EXPR || node->node_type == PT_VALUE) && node->info.expr.op != PT_CASE)
```

Comment:
```suggestion
	  && ((node->node_type == PT_EXPR && node->info.expr.op != PT_CASE) || node->node_type == PT_VALUE))
```
비슷한 비교를 같이 하면 좋을 것 같습니다.

Reply:
6fcdd7c에서 수정했습니다.